src/user_main.c

```diff
@@ -98,6 +102,67 @@ void ICACHE_FLASH_ATTR  gpio16_output_set(uint8 value)
         (READ_PERI_REG(RTC_GPIO_OUT) & (uint32)0xfffffffe) | (uint32)(value & 1)
     );
 }
+void beep(int ms)
+{
+    gpio16_output_set(1);
+    os_delay_us(ms*1000);
+    gpio16_output_set(0);
+}
+
+void enableArc()
+{
+    if (GPIO_INPUT_GET(4) == 0) {
```

Comment:
Is it worth checking the state before changing it?

Reply:
removed